From deaab224b91ea3082c0ce93b6748f44f25cfd881 Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Wed, 1 Feb 2017 22:07:10 +0100
Subject: [PATCH] configure.ac: use pkg-config to check freetype version

the PKG_CHECK_MODULES macro allows to check for a version -- use it
instead of homegrown TUXBOX_APPS_LIB_PKGCONFIG

Signed-off-by: flk <f_l_k@t-online.de>
---
 configure.ac | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5ce417d..d0f8034 100644
--- a/configure.ac
+++ b/configure.ac
@@ -98,19 +98,7 @@ AM_CONDITIONAL(USE_TREMOR, test "$TREMOR" = "yes")
 
 # TUXBOX_APPS_LIB_PKGCONFIG(OPENSSL,openssl)
 TUXBOX_APPS_LIB_PKGCONFIG(CURL,libcurl)
-TUXBOX_APPS_LIB_PKGCONFIG(FREETYPE,freetype2)
-CPPFLAGS+=" $(freetype-config --cflags)"
-AC_MSG_CHECKING([whether FreeType version is 2.5.0 or higher])
-	AC_TRY_CPP([
-		#include <ft2build.h>
-		#include FT_FREETYPE_H
-		#if FREETYPE_MAJOR < 2 || (FREETYPE_MAJOR == 2 && FREETYPE_MINOR < 5)
-		#error Freetype version too low.
-		#endif
-		],
-		[AC_MSG_RESULT(yes)],
-		[AC_MSG_ERROR([Need FreeType library version 2.5.0 or higher])
-	])
+PKG_CHECK_MODULES([FREETYPE], [freetype2 >= 2.5.0], echo "freetype2 >= 2.5.0 found")
 
 # fallback to curl-config (which is ugly for cross-compilation)
 if test -z "$CURL_LIBS" -a -z "$CURL_CFLAGS"; then
@@ -119,6 +107,18 @@ fi
 # fallback to freetype-config (which is ugly for cross-compilation)
 if test -z "$FREETYPE_LIBS" -a -z "$FREETYPE_CFLAGS"; then
 	TUXBOX_APPS_LIB_CONFIG(FREETYPE,freetype-config)
+	CPPFLAGS+="$FREETYPE_CFLAGS"
+	AC_MSG_CHECKING([whether FreeType version is 2.5.0 or higher])
+		AC_TRY_CPP([
+			#include <ft2build.h>
+			#include FT_FREETYPE_H
+			#if FREETYPE_MAJOR < 2 || (FREETYPE_MAJOR == 2 && FREETYPE_MINOR < 5)
+			#error Freetype version too low.
+			#endif
+			],
+			[AC_MSG_RESULT(yes)],
+			[AC_MSG_ERROR([Need FreeType library version 2.5.0 or higher])
+		])
 fi
 
 TUXBOX_APPS_LIB_PKGCONFIG(PNG,libpng)
-- 
2.7.4

